### PR TITLE
Add sprite animation

### PR DESCRIPTION
**Pull request by LumpBloom7**
Animation and Animation sets can be added to a `GameObject` as a component.

A `AnimationSystem` must be used in order for the animation to work. It works best when `AnimationSystem` updates before `RenderSystem`

### CODE REVIEW

**Comment by LumpBloom7** [comment]
I think it is preferable to use a `std::shared_ptr<Animation>` as the component rather than `Animation`. Maybe we should also do the same for other components??

**Review by kasmacioma — COMMENTED** [review]
What about some changes from https://github.com/xapdkop/BloomFramework/commits/beta/Animation (last 3 commits)?
For example, may be it would be better to simplify names AnimationSet's members (mention of animation in each member is unnecessary in **Animation**Set class)

**Comment by LumpBloom7** [comment]
They're all good imo.
Yeah good point about the names.

File: BloomFramework.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27703.2018
+# Visual Studio Version 16
```
Comment (xapdkop) [inline BloomFramework.sln]: Lol, are you using VS 2019 Preview?

**Comment by siew24** [comment]
Yep, he is

**Comment by LumpBloom7** [comment]
> Lol, are you using VS 2019 Preview?

Tried it for a bit. It was nice :)

**Review by kasmacioma — APPROVED** [review]
Everything seems to be ok. At least it works ;)

**Comment by LumpBloom7** [comment]
Well that's good. Last pull of the year! 🎉🎉

**Pull request merged**